components/style/restyle_hints.rs

```diff
@@ -383,7 +415,38 @@ impl DependencySet {
     }
 
     pub fn clear(&mut self) {
-        self.deps.clear();
+        self.common_deps.clear();
+        self.attr_deps.clear();
+        self.state_deps.clear();
+    }
+}
+
+fn test_dependencies<E>(deps: &[Dependency],
```

Comment:
Hm, the free-floating function is still kind of annoying. Could we make a newtype:

struct DependencyList(Vec<Dependency>)

Inside DependencySet and then impl that?